py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/batcher.py

```diff
@@ -34,7 +37,25 @@ def __str__(self) -> str:
 
     def add_file(self, filename: str, filesize: int) -> None:
         """Add a file to this batch"""
-        self.filenames.append(filename)
+        if (
+            self.prefix != ""
+            and filename.startswith(self.prefix)
+            and filename.endswith(self.suffix)
+        ):
+            filename = filename.removeprefix(self.prefix)
+            filename = filename.removesuffix(self.suffix)
+            self.filenames.append(filename)
+
+        else:
+            full_filenames = unpack_filenames(
+                prefix=self.prefix, suffix=self.suffix, filenames=self.filenames
+            )
+            full_filenames.append(filename)
+
+            self.prefix, self.suffix, self.filenames = compress_filenames(
+                full_filenames
+            )
+
```

Comment:
occasionally, we'll have to unpack the filenames and recompress.

there is probably a slightly cheaper way to do this, but i've already tumbled down this rabbit hole long enough.